src/AbstractArray.php

```diff
@@ -491,31 +485,25 @@ public function getRandom()
     /**
      * Pick a given number of random keys/indexes out of this array.
      *
-     * @param int $number The number of keys/indexes (should be > 1 and < $this->count())
+     * @param int $number The number of keys/indexes (should be <= $this->count())
      *
-     * @return array Random keys of array
+     * @return mixed Random keys or key of array
      *
      * @throws \RangeException
-     * @throws \LogicException
      */
     public function getRandomKeys($number)
     {
         $number = (int) $number;
-        $count = $this->count();
-        if ($number < 1 || $number > $count) {
-            throw new \RangeException(sprintf(''
-                . 'The given number "%d" should be greater than or equal to "1" '
-                . 'and less than or equal to "%d" (the number of elements in the array).',
-                $number,
-                $count
-            ));
+        if (!$number || $number === 0) {
+            throw new \LogicException(sprintf('You have to provide a $number greater 0 (%s given)', $number));
```

Comment:
I think we need to get rid of `LogicException` and keep only `RangeException` with `if ($number === 1 || $number === $count) {` statement.